src/main/java/com/esri/samples/map/generate_offline_map_overrides/GenerateOfflineMapOverridesController.java

```diff
@@ -0,0 +1,282 @@
+                      GenerateLayerOption layerOption = layerOptionsIterator.next();
+                      if (layerOption.getLayerId() == layerId) {
+                        switch (layer.getName()) {
+                          // remove the System Valve layer from the layer options if it should not be included
+                          case "System Valve":
+                            if (!systemValvesCheckBox.isSelected()) {
+                              layerOptionsIterator.remove();
+                            }
+                            break;
+                          // remove the Service Connection layer from the layer options if it should not be included
+                          case "Service Connection":
+                            if (!serviceConnectionsCheckBox.isSelected()) {
+                              layerOptionsIterator.remove();
+                            }
+                            break;
+                          // only download hydrant features if their flow is above the minimum specified in the UI
+                          case "Hydrant":
+                            layerOption.setWhereClause("FLOW >= " + minHydrantFlowRateSpinner.getValue());
+                            layerOption.setQueryOption(GenerateLayerOption.QueryOption.USE_FILTER);
+                            break;
+                          //clip water main feature geometries to the extent if the checkbox is selected
+                          case "Main":
+                            layerOption.setUseGeometry(waterPipesCheckBox.isSelected());
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+
+              // create an offline map job with the download directory path and parameters and start the job
+              Path tempDirectory = Files.createTempDirectory("offline_map");
+              job = offlineMapTask.generateOfflineMap(parameters, tempDirectory.toAbsolutePath().toString(), overrides);
+              job.start();
+              generateOfflineMapButton.setDisable(true);
+              cancelJobButton.setDisable(false);
+              job.addJobDoneListener(() -> {
+                if (job.getStatus() == GenerateOfflineMapJob.Status.SUCCEEDED) {
+                  // replace the current map with the result offline map when the job finishes
+                  GenerateOfflineMapResult result = job.getResult();
+                  mapView.setMap(result.getOfflineMap());
+                  graphicsOverlay.getGraphics().clear();
+                  generateOfflineMapButton.setDisable(true);
+                } else if (!"Job canceled.".equals(job.getError().getAdditionalMessage())) {
+                  new Alert(Alert.AlertType.ERROR, job.getError().getAdditionalMessage()).show();
+                }
+                Platform.runLater(() -> progressBar.setVisible(false));
+                generateOfflineMapButton.setDisable(false);
```

Comment:
The first time Generate offline map is clicked, the progress bar loads and the saved offline data are displayed on the screen. As a user, I see the Generate offline map button is enabled again and assume it would create a subset of offline data based on the current view. However, clicking the button results in a null pointer exception at line 153. 

Could disable the Generate Offline Button at this point so that the user isn't tempted to generate a secondary offline map as the UI infers, or refactor the code to allow a re-generation of offline data?